app/services/statements/query.rb

```diff
@@ -1,21 +1,39 @@
 module Statements
   class Query
-    def initialize(lead_provider:)
-      @lead_provider = lead_provider
+    def initialize
+      @scope = Statement.all
+    end
+
+    def belonging_to(lead_provider:)
+      @scope = @scope
+                 .includes(:cohort)
+                 .where(lead_provider:)
+
+      self
+    end
+
+    def by_cohorts(*start_years)
+      if start_years.any?
+        @scope = @scope.where(cohort: { start_year: start_years })
+      end
+
+      self
+    end
+
+    def since(date)
+      if date.present?
+        @scope = @scope.where(updated_at: date..)
+      end
```

Comment:
I guess this is guarding against `nil` dates, but I'm not sure it's necessary, it just leaves the scope open.

```ruby
Statement.where(created_at: nil..).to_sql

#=> "SELECT \"statements\".* FROM \"statements\" WHERE 1=1"
```

Also `since` isn't the clearest name - i.e., is it created since or updated since?

Reply:
+1 for method name being `updated_since`

Reply:
in ECF, we have some concerns around rescuing bad data being added to filters, see this [concern](https://github.com/DFE-Digital/early-careers-framework/blob/main/app/services/api/concerns/filter_updated_since.rb). Could we add something similar for protecting against bad data and returning relevant errors in the [controller](https://github.com/DFE-Digital/early-careers-framework/blob/main/app/controllers/api/api_controller.rb#L4)?